admin/admin.php

```diff
@@ -78,41 +90,60 @@ function rocket_lazyload_options_output() {
-							<label for="lazyload-images"><span class="screen-reader-text"><?php _e( 'Images', 'rocket-lazyload' ); ?></span></label>
-							<span id="describe-lazyload-images" class="rocket-lazyload-label-description"><?php _e( 'Images', 'rocket-lazyload' ); ?></span>
-						</li>
-						<li class="rocket-lazyload-option">
-							<input type="checkbox" value="1" id="lazyload-iframes" name="rocket_lazyload_options[iframes]" <?php checked( rocket_lazyload_get_option( 'iframes', 0 ), 1 ); ?> aria-describedby="describe-lazyload-iframes">
-							<label for="lazyload-iframes"><span class="screen-reader-text"><?php _e( 'Iframes & Videos', 'rocket-lazyload' ); ?></span></label>
-							<span id="describe-lazyload-iframes" class="rocket-lazyload-label-description"><?php _e( 'Iframes & Videos', 'rocket-lazyload' ); ?></span>
-						</li>
+					<ul class="rocket-lazyload-options">
+						<?php foreach ( $options as $slug => $infos ) { ?>
+
 						<li class="rocket-lazyload-option">
-							<input type="checkbox" value="1" id="lazyload-youtube" name="rocket_lazyload_options[youtube]" <?php checked( rocket_lazyload_get_option( 'youtube', 0 ), 1 ); ?> aria-describedby="describe-lazyload-youtube">
-							<label for="lazyload-youtube"><span class="screen-reader-text"><?php _e( 'Replace Youtube videos by thumbnail', 'rocket-lazyload' ); ?></span></label>
-							<span id="describe-lazyload-youtube" class="rocket-lazyload-label-description"><?php _e( 'Replace Youtube videos by thumbnail', 'rocket-lazyload' ); ?></span>
+							<input type="checkbox" value="1" id="lazyload-<?php echo $slug; ?>" name="rocket_lazyload_options[<?php echo $slug; ?>]" <?php checked( rocket_lazyload_get_option( $slug, 0 ), 1 ); ?> aria-labelledby="describe-lazyload-<?php echo $slug; ?>">
+							<label for="lazyload-<?php echo $slug; ?>">
+								<span id="describe-lazyload-<?php echo $slug; ?>" class="rocket-lazyload-label-description"><?php echo $infos['label']; ?></span>
+							</label>
 						</li>
+
+						<?php } ?>
+
 					</ul>
 				</fieldset>
-			<?php
-				settings_fields( 'rocket_lazyload' );
-				submit_button( __( '✓ Save changes', 'rocket-lazyload' ) );
-			?>
-			</form>
-			<div class="rocket-lazyload-cross-sell">
-				<h2 class="rocket-lazyload-cross-sell-title"><?php _e( 'Need To Boost Your Speed Even More?', 'rocket-lazyload' ); ?></h2>
-				<div class="rocket-lazyload-ads">
-					<a href="https://wp-rocket.me?utm_source=wp_plugin&utm_medium=rocket_lazyload"><img src="<?php echo ROCKET_LL_ASSETS_URL; ?>img/wp-rocket@2x.jpg" alt="WP Rocket" width="393" height="180"></a>
-					<?php if ( ! is_plugin_active( 'imagify/imagify.php' ) ) : ?>
-					<a href="https://imagify.io?utm_source=wp_plugin&utm_medium=rocket_lazyload"><img src="<?php echo ROCKET_LL_ASSETS_URL; ?>img/imagify@2x.jpg" alt="Imagify" width="393" height="180"></a>
-					<?php endif; ?>
+			<?php settings_fields( 'rocket_lazyload' ); ?>
+
+			<div class="rocket-lazyload-upgrade">
```

Comment:
Should have a check to see if WP Rocket is already installed and not display the upsell

Reply:
Done. Thank you @Tabrisrp :)